eos-tech-support/eos-write-live-image

```diff
@@ -389,6 +442,29 @@ fi
 
 "$EOS_WRITE_IMAGE" "$OS_IMAGE" "-" > "${DIR_IMAGES_ENDLESS}/endless.img"
 
+if [ "$EXPAND" ]; then
+    if [ -n "$SIZE" ]; then
+        # Expand the image to the provided size
+        USED_SPACE_BYTES=$(df -P -B1 "$DIR_IMAGES_ENDLESS" | awk 'NR==2 {print $3}')
+        EXTRA_BYTES=$(($SIZE-$USED_SPACE_BYTES))
+
+        if [ $EXTRA_BYTES -lt 0 ]; then
+            echo "Cannot expand the image to $SIZE bytes, minimum size is $USED_SPACE_BYTES bytes"
+        fi
```

Comment:
This calculation seems a little funky to me. I think it's trying to avoid shrinking the image file, but it's looking at the currently used bytes in the enclosing filesystem. Those will roughly match, but I think comparing to the size of the image file itself would match better. Furthermore, this doesn't check if the requested size is bigger than the amount of free space. I think it could be a little better like this:
```
IMAGE_BYTES=$(du -b "${DIR_IMAGES_ENDLESS}/endless.img" | cut -f1)
FREE_SPACE_BYTES=$(df -P -B1 "$DIR_IMAGES_ENDLESS" | awk 'NR==2 {print $4}')
if [ -n "$SIZE" ]; then
  if [ $SIZE -lt $IMAGE_BYTES ]; then
    echo "Cannot expand the image to $SIZE bytes, minimum size is $IMAGE_BYTES bytes" >&2
    exit 1
  fi
  EXTRA_BYTES=$(($SIZE - $IMAGE_BYTES))
  if [ $EXTRA_BYTES -gt $FREE_SPACE_BYTES ]; then
    echo "Cannot expand the image to $SIZE bytes, only $FREE_SPACE_BYTES bytes available" >&2
    exit 1
  fi
else
  # Expand to fill all space
  SIZE=$(($IMAGE_BYTES + $FREE_SPACE_BYTES))
fi

if [ $SIZE -gt $IMAGE_BYTES ]; then
  echo "Expanding the image from $IMAGE_BYTES to $SIZE bytes"
  truncate --size "$SIZE" "${DIR_IMAGES_ENDLESS}/endless.img"
fi
```